### PR TITLE
Client Registration test update to handle multiple products

### PR DESCRIPTION
**Pull request by sthirugn**
1. Handles multiple product types including compute nodes and desktops
2. Adds ability to persist the system after test
3. Adds an ability to pass release version

### CODE REVIEW

File: automation_tools/__init__.py [hunk 1]
```diff
@@ -1118,15 +1118,29 @@ def update_basic_packages():
...
+
+        - `None` - default
+        - `compute` - for compute products
+        - `desktop` - for desktop products
+    :param reset_system: Resets the subscription management back to RHSM
+
+    Note:
+    This functions also uses environment variables like ORG, ACTIVATIONKEY,
+    CERTURL, RELVER
+    """
```
Comment (elyezer) [inline automation_tools/__init__.py]: Please add a blank line before finishing a multiline docstring.

File: automation_tools/__init__.py [hunk 2]
```diff
@@ -1118,15 +1118,29 @@ def update_basic_packages():
...
+    :param update_package: Updates the host with latest packages
+    :param product_type: Specifies the product type
+
+        - `None` - default
+        - `compute` - for compute products
+        - `desktop` - for desktop products
+    :param reset_system: Resets the subscription management back to RHSM
+
+    Note:
+    This functions also uses environment variables like ORG, ACTIVATIONKEY,
```
Comment (elyezer) [inline automation_tools/__init__.py]: Can you expand what the env vars are for?

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -1140,13 +1154,15 @@ def client_registration_test(clean_beaker=True, update_packages=True):
     if not cert_url:
         print('You need to install the Candlepin Cert RPM.')
         sys.exit(1)
-
+    # Release version for the client - Optional - no need to error out if not
+    # available
+    rel_ver = os.getenv('RELVER')
```
Comment (elyezer) [inline automation_tools/__init__.py]: Should this task fail if that env var is not defined? If yes, then I recommend that you check at the beginning of the task before doing anything on the server and fail if something is not defined.

File: automation_tools/__init__.py [hunk 4]
```diff
@@ -1154,40 +1170,86 @@ def client_registration_test(clean_beaker=True, update_packages=True):
...
     print('Register/Subscribe using Subscription-manager.')
-    run('subscription-manager register --force'
-        ' --org="{0}"'
-        ' --activationkey="{1}"'
-        ''.format(org, act_key))
+    cmd = ('subscription-manager register --force'
+           ' --org="{0}"'
+           ' --activationkey="{1}"')
+    if rel_ver:
+        cmd = cmd + ' --release="{2}"'
```
Comment (elyezer) [inline automation_tools/__init__.py]: `cmd += ' --release="{2}"'`

File: automation_tools/__init__.py [hunk 5]
```diff
@@ -1154,40 +1170,86 @@ def client_registration_test(clean_beaker=True, update_packages=True):
...
-    run('rpm -q httpd')
-    manage_daemon('start', 'httpd', warn_only=True)
-    print('Stopping "httpd" service and remove "Web Server" group.')
-    manage_daemon('stop', 'httpd', warn_only=True)
-    run('yum groupremove -y "Web Server"', quiet=True)
-    print('Checking if "httpd" is really removed.')
-    run('rpm -q httpd', warn_only=True)
+    print('Installing "Telnet".')
+    run('yum install -y telnet', quiet=True)
+    print('"Telnet" should be installed.')
```
Comment (elyezer) [inline automation_tools/__init__.py]: You can check if telnet was installed or not like:

```python
result = run('yum install -y telnet', quiet=True)
if result.succeeded:
    print('"Telnet" is installed.')
```

For more info http://docs.fabfile.org/en/1.10/api/core/operations.html?highlight=succeeded#fabric.operations.run

**Comment by elyezer** [comment]
ACK pending comments.

File: automation_tools/__init__.py (hunk 3 above)
Comment (sthirugn) [inline automation_tools/__init__.py]: Nope, it is optional.  I commented that in line 1157

File: automation_tools/__init__.py (hunk 4 above)
Comment (sthirugn) [inline automation_tools/__init__.py]: :+1:

**Comment by sthirugn** [comment]
@elyezer Comments addressed

**Comment by elyezer** [comment]
ACK

File: automation_tools/__init__.py [hunk 6]
```diff
@@ -1118,15 +1118,34 @@ def update_basic_packages():
...
+
+    Note:
+    This functions also uses environment variables like
+
+        - `ORG` - (Optional) Org to register.  Default is Default_Organization
+        - `ACTIVATIONKEY` - (Mandatory) Activation key to register
+        - `CERTURL` - (Mandatory) The server cert url to install
+        - `RELVER` - (Optional) Release version to register
+
+    """
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: I expect you want something like this:

```
    """Register client against Satellite 6 and run tests.

    :param clean_beaker: Clean beaker repositories if ``True``.
    :param update_package: Update the host with latest packages if ``True``.
    :param product_type: Set the product type. Legal values:

        ``None``
            default
        ``compute``
            for compute products
        ``desktop``
            for desktop products

    :param reset_system: Reset the subscription management back to RHSM.
        Affected by the following environment variables:

        ``ORG``
            (Optional) Org to register.  Default is Default_Organization
        ``ACTIVATIONKEY``
            (Mandatory) Activation key to register
        ``CERTURL``
            (Mandatory) The server cert url to install
        ``RELVER``
            (Optional) Release version to register

    """
```

Comment (elyezer) [inline automation_tools/__init__.py]: :+1: thanks @Ichimonji10 for your rst-fu skills :smile:

Comment (sthirugn) [inline automation_tools/__init__.py]: Great, ty

Comment (sthirugn) [inline automation_tools/__init__.py]: I like to add an empty line before the env variables. I will update.

File: automation_tools/__init__.py [hunk 7]
```diff
@@ -1154,40 +1182,88 @@ def client_registration_test(clean_beaker=True, update_packages=True):
 
     # Register and subscribe
     print('Register/Subscribe using Subscription-manager.')
-    run('subscription-manager register --force'
-        ' --org="{0}"'
-        ' --activationkey="{1}"'
-        ''.format(org, act_key))
+    cmd = ('subscription-manager register --force'
+           ' --org="{0}"'
+           ' --activationkey="{1}"')
```
Comment (Ichimonji10) [inline automation_tools/__init__.py]: Would it be easier to just interpolate variables right away?

```python
    cmd = (
        'subscription-manager register --force --org="0" --activationkey="{1}"
        .format(org, act_key)
    )
    if rel_ver:
        cmd += ' --release="{0}"'.format(rel_ver)
    run(cmd)
```

**Comment by Ichimonji10** [comment]
ACK pending comment

**Comment by sthirugn** [comment]
agreed

File: automation_tools/__init__.py (hunk 7 above)
Comment (sthirugn) [inline automation_tools/__init__.py]: Agreed

**Comment by sthirugn** [comment]
@Ichimonji10 Comments addressed

**Comment by elyezer** [comment]
Comments addressed. Merging this.

**Comment by elyezer** [comment]
Karma given

**Comment by Ichimonji10** [comment]
:+1:

**Comment by sthirugn** [comment]
Thank you both :)

**Pull request merged**